### PR TITLE
Fixed User Default View in Resource and Shifting

### LINKED ISSUE #9304: Resource tab is always redirecting to list view instead of user default view
**Describe the bug**

When a user clicks on Resource tab from sidebar , it's always redirected to list view instead of changing based on user's default 

**To Reproduce**

1. Click on Resource tab
2. Change the view to board view
3. Now move to another tab and click on Resource tab again
5. Similar issue is present with Shifting Tab too

**Expected behavior**
User default should change based on user latest preference

**Screen Recording**

https://github.com/user-attachments/assets/e1b2ec47-f941-4d53-945a-0b04dd6ac4ba

Click [here](https://github.com/user-attachments/assets/e1b2ec47-f941-4d53-945a-0b04dd6ac4ba) if video is not playing

**Desktop (please complete the following information):**

- OS: Windows
- Browser : Chrome
- Version : Latest

### PR DESCRIPTION
**Pull request by nithish1018**
## Proposed Changes

- Fixes #9304 

Screen Recording:

https://github.com/user-attachments/assets/9d6aa8f7-f667-4011-8bc4-fe61392331c9

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced navigation functions for switching between board and list views, enhancing user experience.
	- Default view settings are now saved in local storage for a more personalized experience.

- **Bug Fixes**
	- Improved code readability and separation of concerns in navigation logic.

- **Style**
	- Updated button components to a new design with primary variant and improved styling.

### CODE REVIEW

File: src/components/Resource/ResourceBoard.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ const COMPLETED = ["COMPLETED", "REJECTED"];
 const ACTIVE = resourceStatusOptions.filter((o) => !COMPLETED.includes(o));
 
 export default function BoardView() {
+  localStorage.setItem("defaultResourceView", "board");
```
[the remark this answers is not in the document]
Comment (nithishsiliveru) [reply src/components/Resource/ResourceBoard.tsx]: Thank you, I've made these changes

File: src/components/Resource/ResourceBoard.tsx [hunk 2]
```diff
@@ -92,10 +93,14 @@ export default function BoardView() {
             currentTab={boardFilter !== ACTIVE ? 1 : 0}
           />
           <div className="flex w-full flex-col gap-2 lg:mr-4 lg:w-fit lg:flex-row lg:gap-4">
-            <ButtonV2 className="py-[11px]" onClick={onListViewBtnClick}>
+            <Button
+              className="h-[45px]"
+              variant={"primary"}
+              onClick={onListViewBtnClick}
+            >
```
Comment (rithviknishad) [inline src/components/Resource/ResourceBoard.tsx]: let's stick with button's default size. 

```suggestion
            <Button
              variant={"primary"}
              onClick={onListViewBtnClick}
            >
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/components/Resource/ResourceBoard.tsx (hunk 2 above)
Comment (nithishsiliveru) [reply src/components/Resource/ResourceBoard.tsx]: Okay, will update it

File: src/components/Resource/ResourceList.tsx [hunk 3]
```diff
@@ -216,10 +219,10 @@ export default function ListView() {
           </div>
 
           <div className="mt-2 flex w-full flex-col gap-2 lg:w-fit lg:flex-row lg:gap-4">
-            <ButtonV2 className="py-[11px]" onClick={onBoardViewBtnClick}>
+            <Button variant={"primary"} onClick={onBoardViewBtnClick}>
               <CareIcon icon="l-list-ul" className="rotate-90" />
```
Comment (Jacobjeevan) [inline src/components/Resource/ResourceList.tsx]: add the margin right similar to what button in ShiftingList has (spacing between the icon and text); same for ResourceBoard.

Comment (nithishsiliveru) [reply src/components/Resource/ResourceList.tsx]: Done, thanks for the heads-up, I missed it

**Review by rithviknishad — APPROVED** [review]

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Pull request merged**